canister/src/validation.rs

```diff
@@ -1,11 +1,11 @@
-use crate::{blocktree::BlockDoesNotExtendTree, state::State, unstable_blocks, Block};
+use crate::{blocktree::BlockDoesNotExtendTree, state::State, unstable_blocks};
 use bitcoin::BlockHeader;
 use ic_btc_validation::HeaderStore;
 
 /// A structure passed to the validation crate to validate a specific block header.
 pub struct ValidationContext<'a> {
     state: &'a State,
-    chain: Vec<&'a Block>,
+    chain: Vec<(&'a BlockHeader, crate::types::BlockHash)>,
```

Comment:
I needed this to use already existing code in ValidationContext, and evermore BlockHeader is sufficient to be stored.  BlockHash is stored to avoid unnecessary recomputation of hash each time we are using BlockHeader.

Reply:
I was thinking about that, that is a good idea.

Reply:
I will leave this as it is, and change it in the follow-up. Do you want me to create a new type or you will do it?